src/sst/core/config.cc

```diff
@@ -41,163 +43,826 @@ using namespace std;
 {
-    debugFile                 = "/dev/null";
-    runMode                   = Simulation::BOTH;
-    libpath                   = SST_INSTALL_PREFIX "/lib/sst";
-    addlLibPath               = "";
-    configFile                = "NONE";
-    stopAtCycle               = "0 ns";
-    stopAfterSec              = 0;
-    timeBase                  = "1 ps";
-    heartbeatPeriod           = "N";
-    partitioner               = "sst.linear";
-    timeVortex                = "sst.timevortex.priority_queue";
-    inter_thread_links        = false;
-    dump_component_graph_file = "";
-    output_partition          = false;
+private:
+    Config& cfg;
+
+public:
+    typedef bool (ConfigHelper::*flagFunction)(void);
+    typedef bool (ConfigHelper::*argFunction)(const std::string& arg);
+
+    ConfigHelper(Config& cfg) : cfg(cfg) {}
+
+    // Verbosity
+    bool incrVerbose()
+    {
+        cfg.verbose_++;
+        return true;
+    }
+
+    bool setVerbosity(const std::string& arg)
+    {
+        errno             = E_OK;
+        unsigned long val = strtoul(arg.c_str(), nullptr, 0);
+        if ( errno == E_OK ) {
+            cfg.verbose_ = val;
+            return true;
+        }
+        fprintf(stderr, "Failed to parse [%s] as number\n", arg.c_str());
+        return false;
+    }
+
+    // num_threads
+    bool setNumThreads(const std::string& arg)
+    {
+        errno              = E_OK;
+        unsigned long nthr = strtoul(arg.c_str(), nullptr, 0);
+        if ( errno == E_OK ) {
+            cfg.world_size_.thread = nthr;
+            return true;
+        }
+        fprintf(stderr, "Failed to parse [%s] as number of threads\n", arg.c_str());
+        return false;
+    }
+
+    // SDL file
+    bool setConfigFile(const std::string& arg)
+    {
+        cfg.configFile_ = arg;
+        return true;
+    }
+
+    // model_options
+    bool setModelOptions(const std::string& arg)
+    {
+        if ( cfg.model_options_.empty() )
+            cfg.model_options_ = arg;
+        else
+            cfg.model_options_ += std::string(" \"") + arg + std::string("\"");
+        return true;
+    }
+
+    // print timing
+    bool enablePrintTiming()
+    {
+        cfg.print_timing_ = true;
+        return true;
+    }
+
+    bool setPrintTiming(const std::string& arg)
+    {
+        bool success      = false;
+        cfg.print_timing_ = parseBoolean(arg, success, "enable-print-timing");
+        return success;
+    }
+
+    // stopAtCycle
+    bool setStopAt(const std::string& arg)
+    {
+        fprintf(stderr, "Program option 'stop-at' has been deprecated.  Please use 'stopAtCycle' instead");
+        cfg.stopAtCycle_ = arg;
+        return true;
+    }
+
+    // stopAtCycle
+    bool setStopAtCycle(const std::string& arg)
+    {
+        /* TODO: Error checking */
+        cfg.stopAtCycle_ = arg;
+        return true;
+    }
+
+    // stop after
+    bool setStopAfter(const std::string& arg)
+    {
+        /* TODO: Error checking */
```

Comment:
I don't think I'll add the checking here because parsing the string is non-trivial.  I will remove the comment and make sure we get a reasonable error at the point where the string is parsed.